kafka/integration_test.go

```diff
@@ -3245,6 +3245,95 @@ func (its *IntegrationTestSuite) TestAdminClient_ListOffsets() {
 
 }
 
+// Test DeleteRecords API which deletes all the records before the specified offset
+// in the particular partition of the specified topic.
+func (its *IntegrationTestSuite) TestAdminClient_DeleteRecords() {
+	t := its.T()
+	bootstrapServers := testconf.Brokers
+	assert := its.Assert()
+
+	// Create a new AdminClient.
+	a := createAdminClient(t)
+	defer a.Close()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// Create a new topic to test the api and produce some messages to that topic
+	topic := fmt.Sprintf("%s-%d", testconf.TopicName, rand.Int())
+
+	topics := []TopicSpecification{TopicSpecification{Topic: topic, NumPartitions: 3, ReplicationFactor: 1}}
+	createTopicResult, createTopicError := a.CreateTopics(ctx, topics)
+	assert.Nil(createTopicError, "Create Topics should not fail.")
+	assert.Equal(createTopicResult[0].Error.Code(), ErrNoError, "Create Topics Error Code should be ErrNoError.")
+
+	p, err := NewProducer(&ConfigMap{"bootstrap.servers": bootstrapServers})
+	assert.Nil(err, "Unable to create Producer.")
+	defer p.Close()
+
+	numMessages := 3
+	numPartitions := 3
+	for i := 0; i < numMessages; i++ {
+		for j := 0; j < numPartitions; j++ {
+			p.Produce(&Message{
+				TopicPartition: TopicPartition{Topic: &topic, Partition: int32(j)},
+				Value:          []byte(fmt.Sprintf("Message-%d", i)),
+			}, nil)
+		}
+	}
+
+	p.Flush(5 * 1000)
+
+	// Delete the records:
+	// Partition 0: upto offset 2
+	// Partition 1: upto the last offset
+	// Partition 2: upto offset 4 (non-existent)
+	// The result will contain the minimum offset available after deletion in that Topic Partiton
+	delRecordsTopicPartitionOffsets := []TopicPartition{
+		{Topic: &topic, Partition: int32(0), Offset: Offset(2)},
+		{Topic: &topic, Partition: int32(1), Offset: OffsetEnd},
+		{Topic: &topic, Partition: int32(2), Offset: Offset(4)},
+	}
+	deleteRes, err := a.DeleteRecords(ctx, delRecordsTopicPartitionOffsets)
+	assert.Nil(err, "Delete Records should not fail")
+	assert.Len(deleteRes.DeleteRecordsResults, 3, "Length of deleteRes.TopicPartitions should be 3")
```

Comment:
Line break and field name change
```suggestion
	assert.Len(deleteRes.DeleteRecordsResults, 3,
		"Length of deleteRes.DeleteRecordsResults should be 3")
```